bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/mapper/SerialisedInputsToActionInputs.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.util.mapper;
+
+import java.math.BigDecimal;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.type.ActionType;
+import org.openhab.core.automation.type.Input;
+import org.openhab.core.i18n.UnitProvider;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a utility class to convert serialised inputs to the Java types required by the {@link Input}s of a
+ * {@link ActionType}.
+ *
+ * @author Laurent Garnier & Florian Hotze - Initial contribution
+ */
+@NonNullByDefault
+public class SerialisedInputsToActionInputs {
```

Comment:
See above. IMO this should be a `Component`.

Reply:
To be checked if all consumers are components.
Do you think I should merge the two classes in one unique component? Or is it fine to have two new components?

Reply:
Creating a component for `SerialisedInputsToActionInputs` seems doable.

Reply:
Helper component created